osh/hub/osh_xmlrpc/worker.py

```diff
@@ -76,6 +79,20 @@ def finish_task(request, task_id):
                 task.fail_task()
 
 
+@validate_worker
+def open_task(request, task_id):
+    response = kobo_open_task(request, task_id)
+
+    if settings.ENABLE_SINGLE_USE_WORKERS:
+        task = Task.objects.get(id=task_id)
+        # Set the worker load to 0. This should avoid getting any new tasks assigned the worker.
+        # Each single use worker should be destroyed once a task completes.
+        task.worker.max_load = 0
```

Comment:
This would cause issues with differential scans. The differential scan task would get assigned, but never start as the `max_load` has been set to `0` before the base scan task is started.

Reply:
In case of differential scans, this should be set to `0` only after "non-base" scan has started.

Reply:
Did you actually hit this problem or is it just your assumption? I don't think this is a problem.  Otherwise, we would be hitting this all the time on production workers where one task would block others to spawn children.

Waiting tasks are not counted towards the actual load of the worker:
https://github.com/release-engineering/kobo/blob/9af65943b183e2904b47855cbc820b8dda2d5643/kobo/hub/models.py#L212

That said, enjoy the rest of the weekend!

Reply:
I hit it while doing a [version-diff-build](https://openscanhub.stg.fedoraproject.org/task/33/) scan. It only started after I reset the `max_load` field manually.

Reply:
Sorry, my previous comment was completely wrong.